src/page/cardAdd/index.jsx

```diff
@@ -0,0 +1,185 @@
+import { useState, useCallback, useEffect } from 'react';
+
+import FormInput from '../../components/common/FormInput';
+import CardPreview from '../../components/CardPreview';
+import Modal from '../../components/common/Modal';
+import CardCompany from '../../components/CardCompany';
+import Header from '../../components/common/Header';
+import Button from '../../components/common/Button';
+import Tooltip from '../../components/common/Tooltip';
+import { ReactComponent as PrevIcon } from '../../assets/prev_icon.svg';
+
+import { validator, checkFullFilled } from './validator';
+import { isObject } from '../../utils';
+import { CRYPTO_STRING, INPUT_MAX_LENGTH } from '../../constants';
+
+import {
+  cardNumberInputInfoList,
+  expiryDateInputInfoList,
+  cardOwnerNameInputInfoList,
+  privacyCodeInputInfoList,
+  cardPasswordInputInfoList,
+  cardCompanyList,
+} from './data';
+
+const initialCardInfo = {
+  company: '',
+  number: {
+    first: '',
+    second: '',
+    third: '',
+    fourth: '',
+  },
+  expiryDate: {
+    month: '',
+    year: '',
+  },
+  ownerName: '',
+  privacyCode: '',
+  password: {
+    first: '',
+    second: '',
+    third: CRYPTO_STRING,
+    fourth: CRYPTO_STRING,
+  },
+  theme: '',
+};
+
+const CardAppPage = () => {
+  const [cardInfo, setCardInfo] = useState(initialCardInfo);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [isfullFilled, setIsFullFilled] = useState(false);
+  const { number, ownerName, expiryDate, company, theme, password, privacyCode } = cardInfo;
+
+  useEffect(() => {
+    if (checkFullFilled(cardInfo)) {
+      setIsFullFilled(true);
+      return;
+    }
+
+    setIsFullFilled(false);
+  }, [cardInfo]);
+
+  const handleChange = ({ target }, item) => {
+    const { name, value } = target;
+
+    if (!validator[item](value, name)) return;
+
+    setCardInfo((prevCardInfo) => {
+      if (isObject(prevCardInfo[item])) {
+        return {
+          ...prevCardInfo,
+          [item]: {
+            ...prevCardInfo[item],
+            [name]: value,
+          },
+        };
+      }
+
+      return {
+        ...prevCardInfo,
+        [item]: value,
+      };
+    });
+  };
+
+  const handleClickCompany = (company, theme) => {
+    setCardInfo((prevCardInfo) => ({
+      ...prevCardInfo,
+      company,
+      theme,
+    }));
+
+    handleModal();
+  };
+
+  const handleModal = useCallback(() => {
+    setModalVisible((prevModalVisible) => !prevModalVisible);
+  }, []);
```

Comment:
참고로 toggle만 한다면 reducer를 통해 간단하게 작성할 수 있어요!
```suggestion
  const [modalVisible, handleModal] = useReducer((visible) => !visible, false);

```

Reply:
헉,,`useReducer`라는 좋은 hook이 존재했군요!! 리액트에 익숙하지 못해서 그런지 전혀 모르고 있었네요 ㅎㅎ.
modal관련해서는 custom hook을 만들기보다는 이렇게 간단하게 `useReducer`를 쓰는게 좋은 것 같아요!

아사님 덕분에 `useReducer`관련해서도 더 알아보게 되었습니다 🙃